File: 3016/solution.java
class Solution {
    public int minimumPushes(String word) {
        Map<Character, Integer> map = new HashMap<>();
        int res = 0;
        int cnt = 0;
        for(int i=0;i<word.length();i++){
            char c = word.charAt(i);
            map.put(c, map.getOrDefault(c,0)+1);
        }
        List<Map.Entry<Character, Integer>> keys = new ArrayList<>(map.entrySet());
        keys.sort(Map.Entry.<Character, Integer>comparingByValue().reversed());
        for(Map.Entry<Character, Integer> entry: keys)
            res+=entry.getValue()*(1+cnt++/8);
        return res;
    }
}

File: 3016/solution.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        counter = Counter(word)
        cnt = 0 
        res = 0 
        for i in sorted(counter.keys(), key = lambda x:-counter[x]):
            res+=counter[i]*(1+cnt//8)
            cnt+=1 
        return res
